docs/data/material/customization/theme-components/theme-components.md

```diff
@@ -56,48 +56,114 @@ Each component is composed of several different parts.
 These parts correspond to classes that are available to the component—see the **CSS** section of the component's API page for a detailed list.
 You can use these classes inside the `styleOverrides` key to modify the corresponding parts of the component.
 
+### Overrides based on props
+
+Each slot of the component's `styleOverrides` support a key of `variants`. Variants are sets of overrides when the specified props are matched.
+
+The definitions are specified in an array with sensitive order, so make sure that the styles that should win are specified last.
+
 ```js
 const theme = createTheme({
   components: {
     MuiButton: {
       styleOverrides: {
-        root: ({ ownerState }) => ({
-          ...(ownerState.variant === 'contained' &&
-            ownerState.color === 'primary' && {
-              backgroundColor: '#202020',
-              color: '#fff',
-            }),
-        }),
+        root: {
+          variants: [
+            {
+              props: { variant: 'dashed' },
+              style: {
+                textTransform: 'none',
+                border: `2px dashed ${blue[500]}`,
+              },
+            },
+            {
+              props: { variant: 'dashed', color: 'secondary' },
+              style: {
+                border: `4px dashed ${red[500]}`,
+              },
+            },
+          ],
+        },
       },
     },
   },
 });
 ```
 
-### Overrides based on props
+If you're using TypeScript, you'll need to specify your new variants/colors, using [module augmentation](https://www.typescriptlang.org/docs/handbook/declaration-merging.html#module-augmentation).
+
+<!-- Tested with packages/mui-material/test/typescript/augmentation/themeComponents.spec.ts -->
 
-You can pass a callback as a value in each slot of the component's `styleOverrides` to apply styles based on props.
+```tsx
+declare module '@mui/material/Button' {
+  interface ButtonPropsVariantOverrides {
+    dashed: true;
+  }
+}
+```
 
-The `ownerState` prop is a combination of public props that you pass to the component + internal state of the component.
+{{"demo": "GlobalThemeVariants.js"}}
+
+The variant `props` can also be defined as a callback.
+This is useful if you want to apply styles when using negation in the condition.
+In other words, applying a different style if a particular property doesn't have a specific value.
```

Comment:
Maybe better?

```suggestion
The variant `props` can also be defined as a callback, allowing you to apply styles based on negation conditions. This is useful for styling when a property does not have a specific value.
```

Reply:
I remove the `negation` to cover all other conditions.